### ISSUE TITLE
react/prop-types does not check nested properties against PropTypes.shape

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1235]
## Summary

A root prop is declared as PropTypes.shape({}), but reading an undeclared nested property receives no diagnostic. ESLint reports data.name. Without a root validator utlint only reports data, which can hide the missing nested check; the validated-root example isolates the actual gap.

Verified with @utoo/lint 0.4.1, ESLint 8.57.1, @typescript-eslint/parser 6.21.0, eslint-plugin-react 7.37.5, Node.js 22.23.1 on macOS arm64.

## Setup

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.1 eslint@8.57.1 @typescript-eslint/parser@6.21.0 eslint-plugin-react@7.37.5
```

`utlint.config.json`:

```json
{
  "rules": {
    "react/prop-types": "warn"
  }
}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": [
    "react"
  ],
  "rules": {
    "react/prop-types": "warn"
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
};
```

Run each example separately using:

```sh
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx
./node_modules/.bin/eslint --no-eslintrc --config .eslintrc.cjs --format=json example.tsx
```

## Reproduction 1

`example.tsx`:

```tsx
import PropTypes from 'prop-types'; export function Example({data}){return <span>{data.name}</span>;} Example.propTypes={data:PropTypes.shape({})};
```

```text
ESLint:
'data.name' is missing in props validation

utlint:
(no diagnostics)
```

## Controls

`example.tsx`:

```tsx
import PropTypes from 'prop-types'; export function Example({data}){return <span>{data.name}</span>;} Example.propTypes={data:PropTypes.shape({name:PropTypes.string})};
```

```text
ESLint:
(no diagnostics)

utlint:
(no diagnostics)
```

`example.tsx`:

```tsx
export function Example({data}){ return <span>{data.name}</span>; }
```

```text
ESLint:
'data' is missing in props validation
'data.name' is missing in props validation

utlint:
'data' is missing in props validation
```

All examples parse successfully. No lint-disable comments or additional rules are involved.
